lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1579,10 +1579,14 @@ public function walkSimpleSelectExpression($simpleSelectExpression)
                 $sql .= $this->walkPathExpression($expr);
                 break;
 
-            case ($expr instanceof AST\AggregateExpression):
+            case ($expr instanceof AST\Functions\AvgFunction):
```

Comment:
Since custom expressions are allowed, why not simply checking against `$expr instanceof FunctionNode`? Any reason why that wasn't done?